### DISCUSSION TITLE
Editing address node to add an amenity

### DISCUSSION THREAD

**Discussion by marcel263** [Zverik/every_door#748] [Q&A]
How can an existing address node be updated to add an amenity instead of having the app create (duplicate) a new node derived from an existing address?

Some territories by comvention need this behavior of adding amenities to existing address nodes.

**Comment by Zverik** [comment]
I strongly advise against adding amenity data to address nodes. The main reason being, it will be easy to lose this address when the amenity is closed or moved. There is an issue open just about that. Conserving address data is more important.

OSM database is not normalized, there are no duplicates between different object types. E.g. see names on road segments.

**Reply by marcel263** [reply]
> I strongly advise against adding amenity data to address nodes. The main reason being, it will be easy to lose this address when the amenity is closed or moved. There is an issue open just about that. Conserving address data is more important.
> 
> OSM database is not normalized, there are no duplicates between different object types. E.g. see names on road segments.

You advise against this, but it makes this app unusable within some communities. Which is too bad. The rationale for a single address node can be debated. I don't understand why the app cannot be adapted to a convention instead of the reverse.

**Reply by Zverik** [reply]
If you invoke a community instead of a single person, please reference the wiki page or another written document that codifies this mapping practice.

**Reply by marcel263** [reply]
> If you invoke a community instead of a single person, please reference the wiki page or another written document that codifies this mapping practice.

For example when editing Netherlands addresses, local editors will have to do extra work; their mapping convention is to tag the address node, duplicating the address node if there are more separate amenities on that address. You can read this on their wiki (have to translate): https://wiki.openstreetmap.org/wiki/Welkom_op_OSM and on their forums: https://community.openstreetmap.org/t/waar-poi-informatie-plaatsen-op-poi-node-bag-node-of-het-gebouw/76969

you practically make the app unattractive to users in that community as they will at least advise against using it, because it adds cleaning up.

**Reply by fred-jonkhart** [reply]
Same for me, a Netherlands user. I am rather a novice and I was corrected by the comunity that I removed address nodes and created duplicate nodes using the ED app.
Do I understand correctly that the app's assumed practice is to create an additional node for an amenity _next_ to the address node?
And as a consequence also delete the node when the amenity no longer exists (Doesn't Exist button => Remove).
